FLT/Assumptions/Mazur.lean

```diff
@@ -0,0 +1,100 @@
+import Mathlib
+import Mathlib.Algebra.Module.Torsion
+import Mathlib.AlgebraicGeometry.EllipticCurve.Group
+import Mathlib.FieldTheory.IsSepClosed
+import Mathlib.RepresentationTheory.Basic
+
+/-!
+
+# Bounds for the torsion of elliptic curves over the rationals.
+
+Let `E` be an elliptic curve over the rational numbers. It is well-known
+that the rational points `E(ℚ)` on the curve form a finitely-generated
+abelian group, and thus `E(ℚ) ≃ ℤʳ × T` where `r` is the algebraic
+rank of the curve and `T`, a finite group, is the torsion subgroup.
+
+Our understanding of the possibilities for the algebraic rank `r` is
+not at all complete. The Birch and Swinnerton-Dyer conjecture claims
+that `r` is the order of vanishing of the L-function `L(E,s)` of `E`
+at `s = 1` (the so-called analytic rank of the curve); this remains open.
+It was certainly my impression in the 1990s that the general belief was
+that `r` should be unbounded (I asked both Rubin and Bombieri this
+question in 1995 and they both seemed to believe it), but since then
+Poonen has given evidence that this might not be the case. Every now
+and then a new elliptic curve is discovered which breaks the world
+record for the highest known rank; at the time of writing, the current
+record-holder is Elkies with a curve of algebraic rank at least 28
```

Comment:
I think it's 29 as of 2024 by [Elkies and Klagsbrun](https://mathoverflow.net/questions/477849/background-for-the-elkies-klagsbrun-curve-of-rank-29)